.github/workflows/auto-create-release-pr.yml

```diff
@@ -9,110 +9,15 @@ jobs:
-          version_v_file=$(fetch_matching_refs "Version-v")
-          release_file=$(fetch_matching_refs "release/")
-
-          # Combine and process: extract {name, semver} for matches, sort desc by semver
-          jq -s 'add | [ .[] | .ref | ltrimstr("refs/heads/") as $name | select($name | test("^Version-v[0-9]+\\.[0-9]+\\.[0-9]+$") or test("^release/[0-9]+\\.[0-9]+\\.[0-9]+$")) | {name: $name, semver: (if $name | test("^Version-v") then $name | ltrimstr("Version-v") else $name | ltrimstr("release/") end) } ] | sort_by( .semver | split(".") | map(tonumber) ) | reverse' "$version_v_file" "$release_file" > all_versions.json
-
-          # Print all found versions (sorted desc)
-          echo "All found versions (sorted desc): $(jq '[ .[].semver ]' all_versions.json)"
-
-          # Print all found matching branches (sorted by semver desc)
-          echo "All found matching branches:"
-          jq -r '.[].name' all_versions.json || echo "No matching branches found."
-
-          # Filter to those with semver strictly lower than current
-          jq --arg semver "$semver" '[ .[] | select( .semver as $v | $semver | split(".") as $c | $v | split(".") as $p | ($p[0] | tonumber) < ($c[0] | tonumber) or (($p[0] | tonumber) == ($c[0] | tonumber) and (($p[1] | tonumber) < ($c[1] | tonumber) or (($p[1] | tonumber) == ($c[1] | tonumber) and ($p[2] | tonumber) < ($c[2] | tonumber)))) ) ]' all_versions.json > filtered_versions.json
-
-          # Print filtered versions (still sorted desc, so [0] is highest lower)
-          echo "Filtered versions (< $semver, sorted desc): $(jq '[ .[].semver ]' filtered_versions.json)"
-
-          # Select the highest lower: first in filtered list
-          if [ "$(jq length filtered_versions.json)" -eq 0 ]; then
-            echo "Error: No versions lower than $semver found. Cannot determine previous-version-ref." >&2
-            exit 1
-          else
-            highest_lower="$(jq -r '.[0].semver' filtered_versions.json)"
-            previous_ref="$(jq -r '.[0].name' filtered_versions.json)"
-            echo "Selected highest version lower than $semver: ${highest_lower}"
-            echo "Selected branch: ${previous_ref}"
-            echo "Passing to previous-version-ref: ${previous_ref}"
-          fi
-          echo "previous_ref=${previous_ref}" >> "$GITHUB_OUTPUT"
-
-          # Print values passed to call-create-release-pr
-          echo "Inputs to call-create-release-pr:"
-          echo "  checkout-base-branch: main"
-          echo "  release-pr-base-branch: stable"
-          echo "  semver-version: ${semver}"
-          echo "  previous-version-ref: ${previous_ref}"
-
-          echo "proceed=true" >> "$GITHUB_OUTPUT"
+          .github/scripts/extract-semver.sh
 
   call-create-release-pr:
     if: ${{ github.ref_type == 'branch' && (startsWith(github.ref, 'refs/heads/Version-v') || startsWith(github.ref, 'refs/heads/release/')) && needs.extract.outputs.proceed == 'true' }}
```

Comment:
Making the assumption that we don't need `proceed` anymore (as it's always set to `true`), and that the other conditions are always true (otherwise .github/scripts/extract-semver.sh script would have thrown an error), do we really need this `if` condition?